db/machine.go

```diff
@@ -49,25 +50,265 @@ func initMachines(tx *sql.Tx) error {
 
 	return err
 }
+//CREATE NEW MACHINE: ADD NEW MACHINE TO THE DATABASE
+func (d DB) CreateMachine(name string, arch string,
+	microarch string, cores int, memoryGB int) error {
+
+	_, err := d.sql.Exec(`
+			INSERT INTO Machines(name, arch, microarch, cores, memory)
+			VALUES (
+				$1,
+				$2,
+				$3,
+				$4,
+				$5
+			)`,
+		name, arch, microarch, cores, memoryGB)
+
+	return err
+}
+//UPDATE MACHINE: MODIFYING MACHINE'S PROPERTIES
+func (d DB) UpdateMachine(id int, row string, arch string, microarch string, cores int, memory int)error{
+	_, err := d.sql.Exec(`
+			UPDATE Machines
+			SET arch = ?, microarch = ?, cores = ?, memory = ?
+			WHERE `+row+` = id
+
+	`,arch, microarch, cores, memory)
+
+	return err
+
+}
+//FILTER MACHINE: SORT LIST BY CORES SIZES
+func (d DB) Sort_Cores()([]Machine, error){
```

Comment:
There are several problems with this method (and the ones that follow it).

1. There is a lot of code duplication
    * each method does the same thing (returns a sorted slice of `Machine` objects) in a slightly different way
    * all of the common code should not be duplicated
    * it would be much better to have a single method (e.g., `GetMachines`) that can sort in a few different ways based on the parameters that are passed to it
2. The names of these methods are not clear
    * if I call `db.Sort_Cores()`, it's not clear that I should receive a `[]Machine` back
    * this would be fixed by merging all of these methods into `GetMachines()`
3. There are mechanically-fixable formatting errors (run `go fmt`)